torch/utils/debug/__init__.py

```diff
@@ -0,0 +1,31 @@
+import torch
+import os
+
+# On loading, check environment variable for default value
+def _set_from_env():
+    value = os.getenv("TORCH_CPP_STACKTRACES", "0")
```

Comment:
Counterintuitively, this means that if I say `TORCH_CPP_STACKTRACES= python foo.py`, then C++ stacktraces will be enabled. That is probably not the intended behavior.

Reply:
At a higher level, I'm a bit ambivalent about implementing it this way. First, you are relying on this Python code being executed prior to the exception being thrown. That may not always be true. Second, we have plenty of precedent of reading out environment variables directly from our codebase, e.g., `ATEN_CPU_CAPABILITY`, and this code breaks with that precedent for... well, I'm not sure what reason this precedent is being broken for.

Reply:
I would have two questions here:
- `ATEN_CPU_CAPABILITY` is read every time you ask for the cpu capability. How fast is reading the env variable and parsing the content? Is it fast enough here?
- `ATEN_CPU_CAPABILITY` is read-only as the user cannot change it with python code. How should this be changed to make it writable? It does not sound perfect to set the env variable every time `set_cpp_stacktraces_enabled` is called.

Or are these things moot because this is not performance critical code (you should not raise errors in a tight loop) and the env variable being the source of truth is good here?

Reply:
> ATEN_CPU_CAPABILITY is read every time you ask for the cpu capability. How fast is reading the env variable and parsing the content? Is it fast enough here?

It's not read every time. `getenv` is called from `compute_cpu_capability` which is called here:

```
CPUCapability get_cpu_capability() {
  static CPUCapability capability = compute_cpu_capability();
  return capability;
}
```

The static variable means that `compute_cpu_capability` is only called once. I would suggest something similar in this case.

> ATEN_CPU_CAPABILITY is read-only as the user cannot change it with python code. How should this be changed to make it writable? It does not sound perfect to set the env variable every time set_cpp_stacktraces_enabled is called.

I don't see why this is something a user would want to do? Like, why does the user need to be able to change it from Python code?

Reply:
The main reason I see for user wanting to change this from python is notebook users that cannot control env variables.
But I guess they can change them just before importing torch and it will work :)